CPAC/QPP/prep_QPP.py

```diff
@@ -0,0 +1,387 @@
+                            for ses_df_tuple in scan_df.groupby('Sessions'):
+                                session = 'ses-{0}'.format(ses_df_tuple[0])
+                                ses_df = ses_df_tuple[1]
+                                if not ses_df_tuple[0] in qpp_dict:
+                                    pre_qpp_dict[ses_df_tuple[0]] = []
+                                pre_qpp_dict[ses_df_tuple[0]].append(ses_df)
+                            for key in pre_qpp_dict:
+                                list_df=pre_qpp_dict[key]
+                                concat_df = list_df[0]
+                                for df in list_df[1:]:
+                                    concat_df=pd.concat(concat_df,df)
+                                qpp_dict[key]=concat_df
+
+                elif group_config_obj.qpp_grpby_strat == "Session":
+                     session='ses-1'
+                     qpp_dict['ses-1'] = scan_df
+                else:
+                     qpp_dict['output_df'] = new_output_df
+        if len(qpp_dict) == 0:
+            err = '\n\n[!]C-PAC says:Could not find match betterrn the ' \
+              'particpants in your pipeline output directory that were ' \
+              'included in your analysis, and the particpants in the phenotype ' \
+                    'phenotype file provided.\n\n'
+            raise Exception(err)
+
+
+    return qpp_dict,inclusion_list
+
+def use_inputs(group_config_file):
+    import os
+    import shutil
+    from CPAC.pipeline.cpac_group_runner import load_config_yml
+    from CPAC.pipeline.cpac_ga_model_generator import create_merged_copefile, create_merge_mask
+
+
+    qpp_dict,inclusion_list = prep_inputs(group_config_file)
+    group_config_obj=load_config_yml(group_config_file)
+    for key in qpp_dict:
+        newer_output_df = qpp_dict[key]
+        pipeline_ID = group_config_obj.pipeline_dir.rstrip('/').split('/')[-1]
+        merge_file = create_merged_copefile(newer_output_df["Filepath"].tolist(), 'merged.nii.gz')
+        merge_mask = create_merge_mask(merge_file, 'merged_mask.nii.gz')
+
+        model_dir = os.path.join(group_config_obj.output_dir,
+                               'cpac_group_analysis',
+                               'CPAC_QPP')
+        out_dir = os.path.join(model_dir,
+                               resource_id,  # nuisance strat to initialize
+                               strat_info,'model_files')
+
+        # series or repeated label == same as qpp scan or sessions list)
+        merge_outdir_string="CPAC_QPP" + "_"+ resource_id
+        merge_outdir = os.path.join(out_dir,merge_outdir_string)
+
+        if not os.path.exists(merge_outdir):
+            os.makedirs(merge_outdir)
+
+        if not os.path.exists(os.path.join(merge_outdir,'merged.nii.gz')):
+            shutil.move(merge_file,merge_outdir)
+
+        if not os.path.exists(os.path.join(merge_outdir,'merged_mask.nii.gz')):
+            shutil.move(merge_mask,merge_outdir)
+
```

Comment:
Why are files being moved around here? Unnecessary I/O is never a good idea. Why not just provide the merge_outdir/merged.nii.gz to the "out_file" input of created_merged_copefile and create_merged_mask instead, at lines 328 and 329?